src/Juvix/Compiler/Core/Extra/Base.hs

```diff
@@ -128,10 +128,10 @@ unfoldType' ty = case ty of
 {- functions on Node -}
 
 mkApps :: Node -> [(Info, Node)] -> Node
-mkApps = foldl' (\acc (i, n) -> mkApp i acc n)
+mkApps m = foldl' (\acc (i, n) -> mkApp i acc n) m . reverse
 
 mkApps' :: Node -> [Node] -> Node
-mkApps' = foldl' mkApp'
+mkApps' n = foldl' mkApp' n . reverse
 
 unfoldApps :: Node -> (Node, [(Info, Node)])
 unfoldApps = go []
```

Comment:
`mkApps` was correct. `unfoldApps` is "wrong" in that it behaves unintuitively: `unfoldApps (App (App f t1) t2) => (f, [t2, t1])`. This should be changed, not `mkApps`.

Reply:
No, wait, it's also fine, you get `(f, [t1, t2])` because you save the right branch first. What was exactly the problem you were complaining about? I don't see any unexpected behaviour in either `mkApps` or `unfoldApps`.

Reply:
done